File: cpp/2611. 老鼠和奶酪.cpp
class Solution {
public:
    int miceAndCheese(vector<int>& reward1, vector<int>& reward2, int k) {
        int n=reward1.size();
        vector<int> result(n);
        int sum=0;
        for(int i=0;i<n;i++){
            sum+=reward2[i];
            result[i]=reward1[i]-reward2[i];
        }
        sort(result.begin(),result.end());
        for(int i=1;i<=k;i++){
            sum+=result[n-i];
        }
        return sum;
    }
};

File: golang/2611. 老鼠和奶酪.go
func miceAndCheese(reward1 []int, reward2 []int, k int) int {
	n := len(reward1)
	diff := make([]int, n)
	ans := 0
	for i, r := range reward2 {
		ans += r
		diff[i] = reward1[i] - r
	}
	sort.Ints(diff)
	for i := 1; i <= k; i++ {
		ans += diff[n-i]
	}
	return ans
}